### PR TITLE
Button with period

### PR DESCRIPTION
**Pull request by yevgenfs**

### CODE REVIEW

File: Src/lib/drivers/button/button.h [hunk 1]
```diff
@@ -80,21 +80,35 @@ button_err_t button_unreg_callback(void);
...
  @return return type of error or ok if work correctly
  */
 button_err_t button_set_timeout(uint32_t timeout_ms);
 
+/**
+ @brief function which return timeout for button with debouncer
+
+ @return return  timeout for button
+ */
+uint32_t button_get_timeout_with_debouncer(void);
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.h:96]: with_debouncer prefix seems extra for this function name

File: Src/lib/drivers/button/button.h [hunk 2]
```diff
@@ -80,21 +80,35 @@ button_err_t button_unreg_callback(void);
...
+
 /**
  @brief function which return how much time button press
 
  @return return how much time button press
  */
 uint32_t button_get_pressed_time(void);
 
+/**
+ @brief function which return how much time button press
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.h]: Please, fix those comments

File: Src/lib/drivers/button/button.c [hunk 3]
```diff
@@ -58,10 +59,10 @@ button_err_t button_unreg_callback(void)
 
 void button_run(void)
 {
-    if (pressed_time_ms >= timeout)
+    if (pressed_time_ms >= timeout )
     {
         objS_buttonCb(e_event_timeout);
-        pressed_time_ms = 0;
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.c:64]: Try to Avoid code commenting in pull requests

File: Src/lib/drivers/button/button.c [hunk 4]
```diff
@@ -7,12 +7,14 @@
 #include "button.h";
 #define MIN_PRESSED_TIME_BORDER 0
 #define MAX_PRESSED_TIME_BORDER 10000
+#define DEBOUNCER 50
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.c]: Debouncer of what? Time, penguins🙂?
For example you can change it to DEBOUNCER_TIME_MS(S)

File: Src/lib/drivers/button/button.c [hunk 5]
```diff
@@ -7,12 +7,14 @@
 #include "button.h";
 #define MIN_PRESSED_TIME_BORDER 0
 #define MAX_PRESSED_TIME_BORDER 10000
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.c:9]: The same as below MS if it in miliseconds

File: Src/lib/UL/spinner.c [hunk 6]
```diff
@@ -125,11 +133,18 @@ e_spinner_err_t spinner_deinit(void)
 
 e_spinner_err_t spinner_run(void)
 {
-    static uint32_t       period                 = 400;
-    static uint32_t       ms_from_last_operation = 0;
-    static int8_t         count                  = 0;
-    static spinner_ctrl_t objL_item              = {0};
-    button_run();
+    static uint32_t       ms_from_last_operation   = 0;
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c]: Looks like it is to many local variables in this function. I assume that it is possible to hide in drivers level or smth like that

Comment (oleh-plotnikov) [reply Src/lib/UL/spinner.c]: Try to keep functions as simple as possible

File: Src/lib/UL/spinner.c [hunk 7]
```diff
@@ -39,22 +39,30 @@ static obj_led_t led_arr[] =
 };
 
 static spinner_state_t spinner_state = e_spinner_state_run;
+static uint32_t        period        = 400;
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c]: The same. Unit type should be specified

File: Src/lib/UL/spinner.c [hunk 8]
```diff
@@ -39,22 +39,30 @@ static obj_led_t led_arr[] =
...
+static _Bool           is_timeout    = 0;
+static queue_t         queue;
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c:44]: Looks not so good to be placed in this layer, any reason for that?

**Review by oleh-plotnikov — CHANGES_REQUESTED** [review]
Should be improved

File: Src/lib/drivers/button/button.h (hunk 2 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.h]: done

File: Src/lib/drivers/button/button.c (hunk 3 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.c:64]: Done

File: Src/lib/drivers/button/button.c (hunk 4 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.c]: Done

File: Src/lib/drivers/button/button.c (hunk 5 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.c:9]: Done

File: Src/lib/UL/spinner.c (hunk 7 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c]: Done

File: Src/lib/UL/spinner.c (hunk 8 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c:44]: Done

File: Src/lib/UL/spinner.c (hunk 6 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c]: Done

**Review by oleh-plotnikov — APPROVED** [review]

**Pull request merged**